ethcore/light/src/on_demand/mod.rs

```diff
@@ -180,6 +185,14 @@ impl Pending {
 		self.net_requests = builder.build();
 		self.required_capabilities = capabilities;
 	}
+
+	// returning no reponse, it will result in an error.
+	// self is consumed on purpose.
+	fn no_response(self) {
+		trace!(target: "on_demand", "Dropping a pending query (no reply)");
+		self.sender.send(Vec::with_capacity(0)).map_err(|_|())
+			.expect("Non used one shot channel");
```

Comment:
I think this will panic if the receiving end has hung up. better to ignore/handle the error than panic